### ISSUE TITLE
[REQUEST] Setting to ignore Dolby Vision.

### ISSUE THREAD

**Issue by Minefield247** [nixgates/plugin.video.seren#533]
I was wandering if it would be possible to add a setting to allow users to ignore Dolby Vision encoded files (they appear to have DV in the title). There appears to be no fallback so if you don't have DV compatible gear you just get the audio. For test purposes most/all of the WanderVision series has DV files. Thanks for the awesome add-on.

**Comment by drinfernoo** [comment]
Commented to keep open.

**Comment by TheLiquidChef** [comment]
I’ve currently added the following lines to my source_utils.py file under the Undesirable Items filter which seems to do the trick:

" dv ", "dovi", "dolby vision", "dolbyvision"

It would be nice if we could get this feature implemented natively in Seren as we currently have to re-add these filters after each update.

**Comment by Liquidrag** [comment]
> I’ve currently added the following lines to my source_utils.py file under the Undesirable Items filter which seems to do the trick:
> 
> " dv ", "dovi", "dolby vision", "dolbyvision"
> 
> It would be nice if we could get this feature implemented natively in Seren as we currently have to re-add these filters after each update.

Where can the source_utils.py be found and what line do you add it to?

**Comment by TheLiquidChef** [comment]
> Where can the source_utils.py be found and what line do you add it to?

I'm running my setup on Mac and wasn't sure were to find the source_utils.py file. I got around it by using the official Kodi Backup add on and then navigating to the backup folder > addons > plugin.video.seren > resources > lib > common > source_utils.py. I then added the code as per the image below.
![Screenshot 2021-12-10 at 10 41 11 AM](https://user-images.githubusercontent.com/84040076/145543759-d169e8ab-c032-4507-8f81-35204818f119.png)

Then I used the Backup add on to restore that backup :)

**Comment by Dealz-416LS** [comment]
> > Where can the source_utils.py be found and what line do you add it to?
> 
> I'm running my setup on Mac and wasn't sure were to find the source_utils.py file. I got around it by using the official Kodi Backup add on and then navigating to the backup folder > addons > plugin.video.seren > resources > lib > common > source_utils.py. I then added the code as per the image below. ![Screenshot 2021-12-10 at 10 41 11 AM](https://user-images.githubusercontent.com/84040076/145543759-d169e8ab-c032-4507-8f81-35204818f119.png)
> 
> Then I used the Backup add on to restore that backup :)

I tried this, but wasn't able to get it to work.

**Comment by TheLiquidChef** [comment]
> I tried this, but wasn't able to get it to work.

Ah sorry to hear that. Have you rebooted Kodi after the backup restoration?

It’s quite an involved work around and works on my side but would be great if we could get a simple toggle. I know Fen has one but I personally prefer using Seren.

**Comment by DJMix-I-Can** [comment]
> > > Where can the source_utils.py be found and what line do you add it to?
> > 
> > 
> > I'm running my setup on Mac and wasn't sure were to find the source_utils.py file. I got around it by using the official Kodi Backup add on and then navigating to the backup folder > addons > plugin.video.seren > resources > lib > common > source_utils.py. I then added the code as per the image below. ![Screenshot 2021-12-10 at 10 41 11 AM](https://user-images.githubusercontent.com/84040076/145543759-d169e8ab-c032-4507-8f81-35204818f119.png)
> > Then I used the Backup add on to restore that backup :)
> 
> I tried this, but wasn't able to get it to work.

Scroll down to "def get_info(release_title):" Now scroll down to where it says "CAM" & add it there as well.

**Comment by Liquidrag** [comment]
> > Where can the source_utils.py be found and what line do you add it to?
> 
> I'm running my setup on Mac and wasn't sure were to find the source_utils.py file. I got around it by using the official Kodi Backup add on and then navigating to the backup folder > addons > plugin.video.seren > resources > lib > common > source_utils.py. I then added the code as per the image below. ![Screenshot 2021-12-10 at 10 41 11 AM](https://user-images.githubusercontent.com/84040076/145543759-d169e8ab-c032-4507-8f81-35204818f119.png)
> 
> Then I used the Backup add on to restore that backup :)

Thank you. On a mac, if you click finder then at the top menu, click "go" then hold "option" on the keyboard, "library" should show up. from there Kodi can be found under Application Support

**Comment by TheLiquidChef** [comment]
> Thank you. On a mac, if you click finder then at the top menu, click "go" then hold "option" on the keyboard, "library" should show up. from there Kodi can be found under Application Support

Ah awesome thanks! That’ll make my life easier when the next update comes around.

**Comment by drinfernoo** [comment]
For a bit more information, the changes you guys are making actually seem to be exploiting a *bug* in the current version of Seren, in which **all** "cam" copies are getting filtered, regardless of any other information about them... You're basically telling Kodi that DV copies are cams.

In a future version there will be a setting for ignoring DV copies, and also the bug I mentioned will be fixed, allowing for cams to be properly filtered.

**Comment by tundrablasen** [comment]
A gesture that ignores Dolby vision is really good. Most of the big files are nowadays DV and picked first. But my tv does not support it so no video for me. That means no autoplay

**Comment by Dealz-416LS** [comment]
Not stale, active feature request.

**Comment by tundrablasen** [comment]
Can this be please included in the normal settings? Thanks

**Comment by TheLiquidChef** [comment]
> Can this be please included in the normal settings? Thanks

Yeah I’m still running into issues having to manually select sources else watch Moon Knight with sound only lol

**Comment by sinanawad** [comment]
Pretty please 😊

**Comment by Liquidrag** [comment]
> > Can this be please included in the normal settings? Thanks
> 
> Yeah I’m still running into issues having to manually select sources else watch Moon Knight with sound only lol

I’m still using your exploit. I don’t watch cams so I can live without them and DV until we get an update. I sure hope Nix is ok.

**Comment by drinfernoo** [comment]
We have this mostly implemented in development, so it should be available whenever Seren gets another update :+1:

**Comment by Liquidrag** [comment]
@drinfernoo glad to hear development is still moving forward. It’s been a minute since the last update and I feared something had gone awry health wise. I look forward to the next update. Many thanks to everyone involved to keeping Seren going.

**Comment by drinfernoo** [comment]
This is now implemented in 2.2.0, by way of new filtering settings 👍

**Comment by TheLiquidChef** [comment]
@drinfernoo Yes, I saw this yesterday! Thanks guys🖤 this is such a help

**Comment by Liquidrag** [comment]
> 👍

Thank you guys for the update! It’s flying 👍🏼👍🏼
